server/mrt_converter/mrt_converter.go

```diff
@@ -0,0 +1,95 @@
+package converter
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/osrg/gobgp/pkg/packet/bgp"
+	"github.com/osrg/gobgp/pkg/packet/mrt"
+)
+
+type update struct {
+	Collector string
+	SeenAt    time.Time
+	PeerAS    uint32
+
+	// Data inside BGP updates.
+	Announced  []*bgp.IPAddrPrefix
+	Withdrawn  []*bgp.IPAddrPrefix
+	Attributes map[bgp.BGPAttrType]bgp.PathAttributeInterface
+}
+
+// extractCollector extracts the collector name from the input file path.
+func extractCollector(filename string) (string, error) {
+	if filename == "" {
+		return "", fmt.Errorf("empty file path")
+	}
+	if filename[0] != '/' {
+		return "", fmt.Errorf("path must starts with '/'")
+	}
```

Comment:
Sure, I added a TODO there.

btw, I changed the parsing algorithm a little bit. When I was looking at your server code and the python client, looks like the path will start with a slash (e.g. `/routeviews.sg/...`). Since our last offline discussion, it would be good to let GCS automatically trigger the converter function; the filepaths will be GCS objects, which cannot contain a preceding slash.

Reply:
ah! ok ,cool. yes.